tests/v2/validation/provisioning/import/import_cluster_test.go

```diff
@@ -0,0 +1,183 @@
+	require.NoError(r.T(), err, "error reading corral configs")
+}
+
+func (r *ImportProvisioningTestSuite) TestProvisioningImportCluster() {
+	corralPackage := corral.PackagesConfig()
+
+	if corralPackage.HasCustomRepo != "" {
+		err := corral.SetCustomRepo(corralPackage.HasCustomRepo)
+		require.Nil(r.T(), err, "error setting remote repo")
+	}
+	newPackages := []string{}
+	if len(corralPackage.CorralPackageImages) == 0 {
+		r.T().Error("No Corral Packages to Test")
+	}
+
+	_, ok := corralPackage.CorralPackageImages[k3sImportPackageName]
+	require.True(r.T(), ok, "Please ensure %s package is in corralPackageImages", k3sImportPackageName)
+
+	var importPackageName string
+
+	for packageName, packageImage := range corralPackage.CorralPackageImages {
+		newPackageName := namegen.AppendRandomString(packageName)
+		if packageName == k3sImportPackageName {
+			importPackageName = newPackageName
+		}
+
+		newPackages = append(newPackages, newPackageName)
+
+		corralRun, err := corral.CreateCorral(r.session, newPackageName, packageImage, corralPackage.HasDebug, corralPackage.HasDebug)
+		require.NoError(r.T(), err, "error creating corral %v", packageName)
+
+		r.T().Logf("Corral %v created successfully", packageName)
+		require.NotNil(r.T(), corralRun, "corral run had no restConfig")
+	}
+
+	importCluster := apiv1.Cluster{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      namegen.AppendRandomString("auto-import"),
+			Namespace: "fleet-default",
+		},
+	}
+
+	decodedKubeConfig, err := corral.GetKubeConfig(importPackageName)
+	require.NoError(r.T(), err)
+
+	config, err := clientcmd.RESTConfigFromKubeConfig(decodedKubeConfig)
+	require.NoError(r.T(), err)
+
+	_, err = clusters.CreateK3SRKE2Cluster(r.client, &importCluster)
+	require.NoError(r.T(), err)
+
+	backoff := wait.Backoff{
+		Duration: 5 * time.Second,
+		Factor:   1.1,
+		Jitter:   0.1,
+		Steps:    10,
+	}
+
+	updatedCluster := new(apiv1.Cluster)
+
+	err = wait.ExponentialBackoff(backoff, func() (finished bool, err error) {
+		updatedCluster, _, err = clusters.GetProvisioningClusterByName(r.client, importCluster.Name, importCluster.Namespace)
+		require.NoError(r.T(), err)
+
+		if updatedCluster.Status.ClusterName != "" {
+			return true, nil
+		}
+
+		return false, nil
+	})
+	require.NoError(r.T(), err)
+
+	logrus.Info(updatedCluster.Status.ClusterName)
+	err = clusters.ImportCluster(r.client, updatedCluster, config)
+	require.NoError(r.T(), err)
+
+	backoff = wait.Backoff{
+		Duration: 5 * time.Second,
+		Factor:   1.1,
+		Jitter:   0.1,
+		Steps:    100,
+	}
+
+	err = wait.ExponentialBackoff(backoff, func() (finished bool, err error) {
+		updatedCluster, _, err = clusters.GetProvisioningClusterByName(r.client, importCluster.Name, importCluster.Namespace)
+		require.NoError(r.T(), err)
+
+		if updatedCluster.Status.Ready {
+			return true, nil
+		}
+
+		return false, nil
+	})
+	require.NoError(r.T(), err)
+
+	podErrors := pods.StatusPods(r.client, updatedCluster.Status.ClusterName)
```

Comment:
Any reason why this specifically is being called and not the other post cluster provisioning checks as well?

Reply:
since its an imported cluster, we just want to make sure the new pods / workloads for the rancher connection finish properly. Since we can't depend on the cluster type I didn't want to try and figure out a generic way to run the verify tests on this